generators/client/templates/angular/src/test/javascript/spec/app/admin/user-management/_user-management.component.spec.ts

```diff
@@ -0,0 +1,146 @@
+<%#
+ Copyright 2013-2017 the original author or authors from the JHipster project.
+
+ This file is part of the JHipster project, see http://www.jhipster.tech/
+ for more information.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+-%>
+<%_
+const tsKeyId = generateTestEntityId(pkType, prodDatabaseType);
+_%>
+/* tslint:disable max-line-length */
+import { ComponentFixture, TestBed, async, inject, fakeAsync, tick } from '@angular/core/testing';
+import { Observable } from 'rxjs/Rx';
+import { Headers } from '@angular/http';
+import { ActivatedRoute, Router } from '@angular/router';
+import { JhiParseLinks, JhiDateUtils, JhiDataUtils, JhiEventManager, JhiAlertService } from 'ng-jhipster';
+
+import { MockAccountService } from './../../../helpers/mock-account.service';
+import { MockPrincipal } from './../../../helpers/mock-principal.service';
+import { MockActivatedRoute, MockRouter } from '../../../helpers/mock-route.service';
+import { <%=angularXAppName%>TestModule } from '../../../test.module';
+import { Principal, AccountService<% if (websocket === 'spring-websocket') { %>, JhiTrackerService<% } %> } from '../../../../../../main/webapp/app/shared';
+import { UserMgmtComponent } from '../../../../../../main/webapp/app/admin/user-management/user-management.component';
+import { UserService, User } from '../../../../../../main/webapp/app/shared';
+
+describe('Component Tests', () => {
+
+    describe('User Management Component', () => {
+        let comp: UserMgmtComponent;
+        let fixture: ComponentFixture<UserMgmtComponent>;
+        let service: UserService;
+        let mockPrincipal: any;
+
+        beforeEach(async(() => {
+            TestBed.configureTestingModule({
```

Comment:
This can be moved into a common file, since it is almost same in all the `user-management-*-spec`

Reply:
There could be some cleanup here but i just want the coverage up now. Its annoying to work with Angular test infra. I already spent too much time on this. If you have time can you try to genaralize the common code :P

Reply:
There is already a test module with some general stuff may be i'll try to move some common stuff over there

Reply:
okay and I will further refactor it after sometime.

Reply:
Thanks for the prompt, moving most to the common test module works